docs/versioned_docs/version-2.4.0/contributing-guide/marketplace/marketplace-setup.md

```diff
@@ -0,0 +1,51 @@
+---
+id: marketplace-setup
+title:  'Marketplace: Setup'
+---
+
+The Marketplace offers custom plugins that can be installed in your ToolJet instance. This guide aims to assist you in creating a new plugin for the ToolJet marketplace.
+
+## Requirements
+- [Node.js](https://nodejs.org/en/download/) **(v18.3.0)**
+- [npm](https://www.npmjs.com/get-npm) **(v8.11.0)**
+
+## Getting started
+
+### Step 1. Enabling the marketplace for your instance
+
+To enable the marketplace for your ToolJet instance, you need to specify the following environment variables in your **`.env`** file:
+
+#### Marketplace feature enable
+
+Use this environment variable to enable/disable the feature that allows users to use the [marketplace](/docs/marketplace).
+
+| variable                   | value             |
+| -------------------------- | ----------------- |
+| ENABLE_MARKETPLACE_FEATURE | `true` or `false` |
+
+#### Enable Marketplace plugin developement mode
+
+The use of this environment variable facilitates plugin development by enabling automatic builds whenever package changes occur, thus simplifying the development process. Moreover, it also incorporates a reload button that retrieves all the recent local modifications from the file system for installed plugins, making it a valuable feature for improving the overall development experience.
+
+| variable                   | value             |
+| -------------------------- | ----------------- |
+| ENABLE_MARKETPLACE_DEV_MODE | `true` or `false` |
+
+
+Please note that the marketplace is not enabled by default. After updating the variable, restart your ToolJet instance. 
+
+For information on running ToolJet on your local machine, please refer to the instructions provided **[here](/docs/category/contributing-guide)**. You can access the marketplace by navigating to the **'/integrations'** route.
+
+### Step 2: Installation of tooljet-cli
+
+To create and publish plugins for the ToolJet marketplace, you will need to utilize **[tooljet-cli](https://www.npmjs.com/package/@tooljet/cli)**. This can be installed via npm by entering the following command:
```

Comment:
```suggestion
In order to manage plugins for the ToolJet marketplace, including creating, updating, and deleting, you will need to utilize **[tooljet-cli](https://www.npmjs.com/package/@tooljet/cli)**. This can be installed via npm by entering the following command:
```